assets/js/hooks/useMonitorInternetConnection.test.js

```diff
@@ -0,0 +1,179 @@
+			},
+		},
+	};
+
+	beforeEach( () => {
+		registry = createTestRegistry();
+
+		store = registry.stores[ CORE_UI ].store;
+	} );
+
+	afterEach( () => {
+		mockOnlineStatus();
+	} );
+
+	beforeAll( () => {
+		originalNavigatorOnline = Object.getOwnPropertyDescriptor(
+			navigator,
+			'onLine'
+		);
+	} );
+
+	afterAll( () => {
+		Object.defineProperty( navigator, 'onLine', originalNavigatorOnline );
+	} );
+
+	it( 'should set online status correctly', () => {
+		fetchMock.getOnce( healthCheckEndpoint, {
+			body: healthCheckResponse,
+		} );
+
+		renderHook( () => useMonitorInternetConnection(), { registry } );
+
+		expect( store.getState().isOnline ).toBe( true );
+	} );
+
+	it( 'should set offline status correctly', () => {
+		mockOnlineStatus( false );
+
+		renderHook( () => useMonitorInternetConnection(), { registry } );
+
+		expect( store.getState().isOnline ).toBe( false );
+	} );
+
+	it( 'should correctly change connection status when going offline then online', async () => {
+		fetchMock.get( healthCheckEndpoint, {
+			body: healthCheckResponse,
+		} );
+
+		renderHook( () => useMonitorInternetConnection(), {
+			registry,
+		} );
+
+		// Wait for fetch to complete.
+		await waitForTimeouts( 100 );
+
+		// Initial status should be online.
+		expect( store.getState().isOnline ).toBe( true );
+
+		// Simulate going offline.
+		await act( async () => {
+			mockOnlineStatus( false );
+			global.window.dispatchEvent( new Event( 'offline' ) );
+			// Wait for fetch to complete.
+			await waitForTimeouts( 100 );
+		} );
+
+		expect( store.getState().isOnline ).toBe( false );
+
+		// Simulate going back online.
+		await act( async () => {
+			mockOnlineStatus();
+			global.window.dispatchEvent( new Event( 'online' ) );
+			// Wait for fetch to complete.
+			await waitForTimeouts( 100 );
+		} );
+
+		expect( store.getState().isOnline ).toBe( true );
+		expect( fetchMock ).toHaveFetched( healthCheckEndpoint );
+	} );
+
+	it( 'should check online status in correct interval when online', () => {
+		jest.useFakeTimers();
+
+		fetchMock.get( healthCheckEndpoint, {
+			body: healthCheckResponse,
+		} );
+
+		renderHook( () => useMonitorInternetConnection(), { registry } );
+
+		expect( store.getState().isOnline ).toBe( true );
+
+		// The initial fetch will happen immediately.
+		expect( fetchMock ).toHaveFetchedTimes( 1 );
+
+		jest.advanceTimersByTime( 119999 );
+
+		expect( fetchMock ).toHaveFetchedTimes( 1 );
+
+		jest.advanceTimersByTime( 1 );
+
+		// The second fetch will happen after 120000ms.
+		expect( fetchMock ).toHaveFetchedTimes( 2 );
+	} );
+
+	it( 'should check online status in correct interval when offline', () => {
+		jest.useFakeTimers();
+
+		fetchMock.get( healthCheckEndpoint, {
+			body: healthCheckResponse,
+		} );
+
+		mockOnlineStatus( false );
+
+		renderHook( () => useMonitorInternetConnection(), { registry } );
+
+		expect( store.getState().isOnline ).toBe( false );
+
+		jest.advanceTimersByTime( 15000 );
```

Comment:
I'd suggest a slightly different tactic here - in order to positively confirm the interval, we could toggle the online mode just before the interval is up. For example:

```js
	it( 'should check online status in correct interval when offline', async () => {
		jest.useFakeTimers();

		fetchMock.get( healthCheckEndpoint, {
			body: healthCheckResponse,
		} );

		mockOnlineStatus( false );

		renderHook( () => useMonitorInternetConnection(), { registry } );

		expect( store.getState().isOnline ).toBe( false );

		// The interval should be 15000ms when offline.
		jest.advanceTimersByTime( 14999 );

		// Enable online status so we can verify the interval via the fetch.
		mockOnlineStatus( true );

		// No fetch should happen when offline until the interval is reached.
		expect( fetchMock ).toHaveBeenCalledTimes( 0 );

		await act( async () => {
			jest.advanceTimersByTime( 1 );
			await new Promise( ( resolve ) => resolve() );
		} );

		expect( fetchMock ).toHaveFetchedTimes( 1 );
	} );
```

What do you think?

Reply:
That's great idea! Thanks, didn't think of that